wandb/sdk/launch/agent/agent.py

```diff
@@ -596,13 +600,30 @@ def _check_run_finished(self, job_tracker: JobAndRunStatusTracker) -> bool:
         try:
             run = job_tracker.run
             status = run.get_status().state
-            if status in ["stopped", "failed", "finished"]:
+
+            # TODO change these statuses to an enum
+            if status in ["stopped", "failed", "finished", "preempted"]:
                 if job_tracker.is_scheduler:
                     wandb.termlog(f"{LOG_PREFIX}Scheduler finished with ID: {run.id}")
                 else:
                     wandb.termlog(f"{LOG_PREFIX}Job finished with ID: {run.id}")
                 with self._jobs_lock:
                     job_tracker.completed_status = status
+                if status == "preempted" and job_tracker.entity == self._entity:
```

Comment:
This seems unnecessarily nested and logically different from the work above. I'd move this into it's own if statement, outside of the containing if statement